kobotoolbox/kobo-install
# Allow Frontend devs to skip container static files out-of-sync checks on start.

Closes #45 

Because some `npm` binaries are OS dependant, it's not possible to run 
`npm install` from one OS (i.e. from within the container - Linux) and run `npm run build` from another OS (i.e. macOS). 

To avoid this, a new environment variable `FRONTEND_DEV_MODE` has been introduced. If it is equal to `host`, the build within the container is skipped even the init script detects that static files are out-of-sync. 

With fresh install, the developer still has to run `npm install` because `node_modules` within the container is not exposed to host (anymore). 

To test this PR: 
- Clone `KPI` and use branch `python3-upgrade`
- Clone `KoBoCat` and use branch `2155_kpi_two_databases`
- Run `python run.py --setup` 
   1. Choose to install `kobo-docker` to new location
   2. Choose to run `npm` from within the container first
   3. Wait the app to be up & running and ensure you can login with `super_admin` and `js/css` is working.
- Run `python run.py --setup` again
   1. Choose to run `npm` locally
   2. Wait the app to be up & running
   3. Run `npm install` locally (you may need to remove `node_modules` first)
   4. Run `npm run watch` locally
   5. Try to login with `super_admin` and see if `js/css` is working.
